src/track-type-dependency.js

```diff
@@ -9,8 +10,21 @@ export default function trackTypeDependency(typeName) {
   types[typeName] = true;
 }
 
+export function trackFieldDependency(typeName, fieldName) {
+  if (!tracking) {
+    return;
+  }
+
+  if (!fields[typeName]) {
+    fields[typeName] = {};
+  }
+
+  fields[typeName][fieldName] = true;
```

Comment:
The current usage of reduce is a true fold though. This implementation would actually be a filter, which is also a fold, but messier. I dunno. This approach is consistent with what's done types, and it's also a fold on the types and doesn't require iterating the underlying array. I think I'd need two reduces to do it with an array, one for the container, and one per type.